Code/Tools/SceneAPI/SceneBuilder/Importers/AssImpBoneImporter.cpp

```diff
@@ -106,37 +67,39 @@ namespace AZ
                     return Events::ProcessingResult::Ignored;
                 }
 
-                bool isBone = false;
-                
-                {
-                    AZStd::unordered_map<AZStd::string, const aiBone*> boneLookup;
-                    MakeBoneMap(scene, boneLookup);
+                AZStd::unordered_multimap<AZStd::string, const aiBone*> boneByNameMap;
+                FindAllBones(scene, boneByNameMap);
```

Comment:
This is a great change and we do need to unify all those places that have different version. 
But I questions why this function is called here because this function "importBones" is called per nodes. FindAllBones function is looping through the scenes which would yield the same result per scene. I think we could call this one time per scene and pass in the boneMap, instead of calling this function per bone.
P.S This problem seems to exist before this change already.

Reply:
Yeah this is a known issue, there's a jira for looking into it.  We haven't found a good way of running something per-scene yet.

Reply:
In this case, should be possible to cache the bone map onto the sourceScene itself?

Reply:
The solution to this is this post-process: https://github.com/assimp/assimp/blob/master/code/PostProcessing/ArmaturePopulate.h

This will require us to recompile the AssImp lib though and will be done later.